docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -106,20 +122,60 @@ local c = a + b;
 }
 ----
 
-* Put one space or line break after opening curly brackets and before closing curly brackets.
+* Put one space or line break after `{` and before `}`.
 +
 [source,jsonnet]
 ----
 // CORRECT
 local foo = { hello: "world" };
-local bar = {
-  hello: "world",
-};
+local bar =
+  {
```

Comment:
Not very happy about this one. It creates more indent and is less readable in my opinion...

Reply:
Ah sorry, misunderstood the rule.

Reply:
I agree

Reply:
It's not noticeable with the simple example, but for more complex examples, I find that this style actually massively improves readability, as the block is visually much more clear.

I'm fine with allowing both, but I find the variant with the opening curly brace just after the `=` much less readable than the variant which has the opening curly brace on it's own line, especially when the object becomes bigger, see for example https://github.com/projectsyn/component-backup-k8up/blob/20cb55a5f41259577ef7c87277372d7be47db742/lib/backup-k8up.libjsonnet#L207-L228

In any case the indentation is absolutely required for visual clarity when the value of a local variable consists of multiple statements. Consider the following two examples:

```
local bar = 
local baz = 3;
{ bar: baz };
```

```
local bar = 
  local baz = 3;
  { bar: baz };
```

Or to completely over-do it (this still compiles just fine):

```
local bar = local baz = 3;
{ bar: baz };

{ foo: bar }
```
```
$ jsonnet test.jsonnet
{
   "foo": {
      "bar": 3
   }
}
```

Reply:
After discussion with @srueg we think we should have the following for variables holding a single object.

```
// PREFERRED
local foo = { bar: 1 };
local foo = {
  bar: 1,
};

// ACCEPTABLE
local foo =
  {
    bar: 1,
  };
```

If the value of a local or field is a larger block we should have the following guide lines:

```
// PREFERRED
local foo = {
  local bar = 3;
  baz: bar,
};
local foo(x) =  {
  local bar(y) = if y < 5 then {
    baz: '<5'
  } else {
    baz: '>=5'
  };
  result: bar(x),
};

// ACCEPTABLE
local foo =
  local bar = 3;
  {
      baz: bar,
  };

// DISALLOWED
local foo =
local bar = 3;
{
  baz: bar,
};
local foo =
{
  local bar = 3;
  baz: bar,
};
local foo(x) =
  local bar(y) = if y < 5 then
  {
    baz: '<5'
  }
  else { baz: '>=5' };
  {
    result: bar(x),
  };
```

Reply:
Defining local variables within an object is not possible. This resulted in the rule to start objects on the same line as the variable itself. Everything else is automatically defined by the formatter.

On top of this, I discussed conditionals with @simu. We also came up with something related to them that is aligned with how object assignments are defined.